### PR TITLE
feat: deprecate `LegacyTokenData`

### PR DESCRIPTION
**Pull request by 0xAlec**
**What changed? Why?**
- change the upstream API to return `Token` data shape

**Notes to reviewers**

**How has it been tested?**
- `yarn test`

### CODE REVIEW

File: .changeset/great-schools-smell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@coinbase/onchainkit": minor
+---
+
+**feat**: deprecate LegacyTokenData
```
Comment (cpcramer) [inline .changeset/great-schools-smell.md:5]:
```suggestion
- **feat**: deprecate LegacyTokenData. By @0xAlec #478 
```

File: src/token/core/getTokens.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { ListSwapAssets } from '../../definitions/swap';
-import { LegacyTokenData, GetTokensOptions, GetTokensResponse, GetTokensError } from '../types';
+import { Token, GetTokensOptions, GetTokensResponse, GetTokensError } from '../types';
```
Comment (cpcramer) [inline src/token/core/getTokens.ts:2]:
```suggestion
import type { Token, GetTokensOptions, GetTokensResponse, GetTokensError } from '../types';
```

File: src/token/core/getTokens.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { ListSwapAssets } from '../../definitions/swap';
-import { LegacyTokenData, GetTokensOptions, GetTokensResponse, GetTokensError } from '../types';
+import type { Token, GetTokensOptions, GetTokensResponse, GetTokensError } from '../types';
```
Comment (Zizzamia) [inline src/token/core/getTokens.ts:2]: I can polish this later, but in general `types` are at the end.

File: src/token/types.ts [hunk 4]
```diff
@@ -1,21 +1,6 @@
 // 🌲☀️🌲
 import { Address } from 'viem';
 
-// The raw response from the Swap API
-// Contains legacy fields that have been renamed in the OnchainKit Token type
-export type LegacyTokenData = {
```
Comment (Zizzamia) [inline src/token/types.ts:6]: Brilliant

**Pull request merged**